compiler/test/dotty/tools/backend/jvm/InlineBytecodeTests.scala

```diff
@@ -281,4 +281,51 @@ class InlineBytecodeTests extends DottyBytecodeTest {
 
     }
   }
+
+    // Testing that a is not boxed
+    @Test def i4522 = {
+    val source = """class Foo {
+                  |  def fun: Int = {
+                  |    var a = 10
+                  |    rewrite def f(arg: => Unit) = {
+                  |      a += 1
+                  |      arg
+                  |    }
+                  |
+                  |    f {
+                  |      return a
+                  |    }
+                  |    a
+                  |  }
+                  |}
+                 """.stripMargin
+
+    checkBCode(source) { dir =>
+      val clsIn      = dir.lookupName("Foo.class", directory = false).input
+      val clsNode    = loadClassNode(clsIn)
+
+      val fun = getMethod(clsNode, "fun")
+      val instructions = instructionsFromMethod(fun)
+      val expected =
+        List(
+          IntOp(BIPUSH, 10)
+          , VarOp(ISTORE, 1)
+          , VarOp(ILOAD, 1)
+          , Op(ICONST_1)
+          , Op(IADD)
+          , VarOp(ISTORE, 1)
+          , VarOp(ILOAD, 1)
+          , Op(IRETURN)
+          , Label(8)
+          , FrameEntry(0, List(), List("java/lang/Throwable"))
+          , Op(ATHROW)
+          , Label(11)
+          , FrameEntry(4, List(), List("java/lang/Throwable"))
+          , Op(ATHROW)
```

Comment:
The 6 lines above should probably not be generated.

There is something fishy going on with the `return a` statement.

Reply:
You are right. I will investigate that in a separate issue because it is unrelated to this one.

Reply:
Updated and opening a new issue for the exception issue investigation. Merci!

Reply:
So @allanrenucci @nicolasstucki I think the observed bytecode is on par with the spec.

The minimal test case is this:

```scala
class Test {
  def fun: Int = {
    var a = 10
    { // throw is generated only when return is nested
      return a
    }
  }
}
```

According to the spec for [return expressions](https://www.scala-lang.org/files/archive/spec/2.12/06-expressions.html#return-expressions):

> Returning from the method from within a nested function may be implemented by throwing and catching a scala.runtime.NonLocalReturnException

So, I think this was the correct behavior (although it is just a nested block and not a nested function).

Reply:
Yes we should open an issue. `scalac` does not emit the `athrow` instruction in this example.

Also I don't think it has anything to do with non local return. The tree for `Test::fun` is left unchanged by the `NonLocalReturns` phase. This seems to be an issue with the backend

Reply:
Tracking in #5064 hereafter. Thx @allanrenucci